addons/project/models/project_project.py

```diff
@@ -94,7 +94,7 @@ def _get_default_favorite_user_ids(self):
 
     @api.model
     def _read_group_stage_ids(self, stages, domain, order):
-        return self.env['project.project.stage'].search([], order=order)
+        return self.env['project.project.stage'].search([('company_id', 'in', (*self._context.get('allowed_company_ids', []), False))], order=order)
```

Comment:
The domain is really needed since we have the ir.rule for multi-company that you added for that model? 🤔

Reply:
It is useless indeed